### PR TITLE
Expand explanation of how to configure HLS

### PR DESCRIPTION
**Pull request by michaelpj**
This topic tends to confuse people, which is unsurprising since there
are several parts which can in principle be configured separately. I've
tried to explain this a bit better.

I also tried to give some brief documentation on the language server
settings that we provide. In the process I found that a few of them seem
to be currently unused, so I'm not sure what to say about them.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -253,44 +246,78 @@ If your desired ghc has been found, you use it to install haskell-language-serve
...
+That means that the way in which you configure the settings will depend on the client.
 
-This configuration is done via the LSP settings you provide with your editor and/or LSP client.
-Some examples include:
-* in VSCode/VSCodium this is done via the `Settings` tab
-* with `LanguageClient-neovim` you can point the [`g:LanguageClient_settingsPath`](https://github.com/autozimu/LanguageClient-neovim/blob/0e5c9546bfddbaa2b01e5056389c25aefc8bf989/doc/LanguageClient.txt#L221)
-  variable to the file in which you want to keep your LSP settings
+Most clients (editors) already have an opinion about how settings should be configured!
+For example, in VS Code you use the graphical Settings tab or `settings.json`, whereas in Emacs you use customization variables.
+In the [editor configuration section](#configuring-your-editor) we give some pointers for popular editors, but you should consult the documentation for your specific editor if you have trouble.
```
Comment (googleson78) [inline README.md:256]: The "pointers" are missing, I think, so the first part of the sentence is a bit confusing.

File: README.md [hunk 2]
```diff
@@ -648,3 +676,12 @@ To do every time you changed code and want to test it:
   - With Stack: `stack build haskell-language-server:exe:haskell-language-server`
 - Restart HLS
   - With VS Code: `Haskell: Restart Haskell LSP Server`
+
+### Adding support for a new editor
+
+Adding support for new editors is fairly easy if the editor already has good support for generic LSP-based extensions.
+In that case, there will likely be an editor-specific support system for this (like `lsp-mode` for Emacs).
+This will typically include instructions for creating a new language server client.
```
Comment (googleson78) [inline README.md:684]: I don't think this is true. You're not creating a new lsp client - that would mean the entire idea lsp is based on is obsolete. 

And some editors (e.g. (n)vim, kakoune, sublime (as far as I can tell)) directly use their lsp clients without Haskell specific "wrappers" - you only need to do configuration.

File: README.md (hunk 1 above)
Comment (michaelpj) [reply README.md:256]: Hm yes, it's not very explicit, let's fix that.

File: README.md (hunk 2 above)
Comment (michaelpj) [reply README.md:684]: Okay, I'll make this less prescriptive.

File: README.md (hunk 1 above)
Comment (googleson78) [reply README.md:256]: Ah, sorry! My original comment was kind of dumb in retrospect, I hadn't noticed the markdown formatting!
In any case, thanks for adding more stuff, I would "resolve" this if I was more adequate at github!

File: README.md (hunk 2 above)
Comment (googleson78) [reply README.md]: Cool, thanks!

**Review by jneira — APPROVED** [review]
Many thanks, the configuration section looks much more neat and clean

**Pull request merged**